[ALT-817-visual] Add github documentation for the visual editor package (#604)
contentful/experience-builder: packages/visual-editor/README.md

- **Browser Limitations**: Currently only Chrome supports both Click/Drag and Drop while Safari and Firefox only allows for Click and Drop.
- **Trees**: Experiences store their own version of the tree that is separate from the source of truth which is stored as a Contentful entry. This is because by storing its own tree version it can provide ui optimizations and manipulations to the tree to maximize performance related tasks such as when performing drag and drop or re-ordering of components. A concrete example where this optimization is necessary is when you drop a component onto the canvas. Without its own version, the component would flicker to its original pre-drag spot before being placed on the intended area based on the mouse's location. 
- **Updating the source of truth**: On drag end, the iFrame will update its own tree before sending a post message to the parent frame in `useCanvasInteractions`. Then the parent frame will update its tree which is the source of truth before sending a post msg back in  `useEditorSubscriber` to the iFrame ensuring that both trees are the same and any discovered discrepancies are resolved.

> - **Pangea DND**: Library providing HOC (higher order components) that Experiences leverages for Drag and Drop. The two components being used from this library is the `Dropzone` and `Draggable` component.

Reviewer: "Pangea DND" seems like slang to me. Should we mention the library name as `@hello-pangea/dnd`?

Reviewer: > The two components being used from this library is the `Dropzone` and `Draggable` component.

We're using `Draggable` and `Droppable` from this library, and `Dropzone` is a custom component.

```suggestion
- **@hello-pangea/dnd**: Library providing drag and drop capabilities for the Experiences Canvas. The two components being used from this library are the `Draggable` and `Droppable` components.
```

- **Dropzones**: Div wrappers that represent areas for other components to be dropped in. Note that only one can be activated at a time via a hitbox and is a separate div from `Draggables`.
- **Draggables**: Div wrappers that allows component to be picked up and dropped. Note that this is a separate div from `Dropzones`.
- **SimulateDnD**: Custom built DND component which is needed as a visual trick for dragging a component from the sidebar onto the iFrame. Since the drag is initiated in the parent frame when clicking and dragging a new component, a simulated action in the iFrame is also triggered so the iFrame is in a drag state just like the parent frame. Then when the mouse crosses over from the parent frame into the iFrame, post messaging is required where the iFrame will send mouse coordinates to the parentFrame so that the dragged component icon is placed in the correct position while the iFrame will trigger the appropriate hitboxes and dropzones since all the hitboxes and dropzones live in the iFrame. 

> - **Component Wrapper** There are several types of div wrappers such as `Dropzones` and `Draggables`. However, there is a third wrapper that is provided as a custom boolean option with `Custom Components`. When this boolean is true, all style options selected from the ui will be applied to this wrapping div. This is done so that the user's css styles on their own components are preserved and shielded from the style options chosen in the ui. 

Reviewer: Nit pick - Line 19 is missing the `:` after `Component Wrapper` that other bullets have
